src/main/java/com/autotune/analyzer/recommendations/engine/RecommendationEngine.java

```diff
@@ -210,14 +231,18 @@ public KruizeObject prepareRecommendations(int calCount) {
         Map<String, KruizeObject> mainKruizeExperimentMAP = new ConcurrentHashMap<>();
         Map<String, Terms> terms = new HashMap<>();
         ValidationOutputData validationOutputData;
-        interval_end_time = Utils.DateUtils.getTimeStampFrom(KruizeConstants.DateFormats.STANDARD_JSON_DATE_FORMAT,
-                intervalEndTimeStr);
-        setInterval_end_time(interval_end_time);
+        Timestamp interval_start_time = null;
+        if (intervalEndTimeStr != null) {       //TODO remove this check and avoid same if across this flow
+            interval_end_time = Utils.DateUtils.getTimeStampFrom(KruizeConstants.DateFormats.STANDARD_JSON_DATE_FORMAT,
+                    intervalEndTimeStr);
+            setInterval_end_time(interval_end_time);
+        }
         KruizeObject kruizeObject = createKruizeObject();
         if (!kruizeObject.getValidation_data().isSuccess())
             return kruizeObject;
-        // continue to generate recommendation when kruizeObject is successfully created
         setKruizeObject(kruizeObject);
+        mainKruizeExperimentMAP.put(kruizeObject.getExperimentName(), kruizeObject);
```

Comment:
This Line should not affect ROS worklow it just update the map which is used to save recommendations